client/asset/eth/eth.go

```diff
@@ -394,7 +394,9 @@ var _ asset.DynamicSwapOrRedemptionFeeChecker = (*TokenWallet)(nil)
 var _ asset.BotWallet = (*assetWallet)(nil)
 
 type baseWallet struct {
-	ctx        context.Context // the asset subsystem starts with Connect(ctx)
+	// The asset subsystem starts with Connect(ctx). This ctx will be initialized
+	// in parent ETHWallet once and re-used in child TokenWallet instances.
+	ctx        context.Context
```

Comment:
It seems `baseWallet.ctx` is the only field on `baseWallet` suffering from this kind of race.